tidy3d/components/data.py

```diff
@@ -397,39 +350,45 @@ def colocate(self, x, y, z) -> xr.Dataset:
                     coord_vals = coord_val_map[coord_name]
                     centered_data_array = centered_data_array.interp(**{coord_name: coord_vals})
             centered_data_dict[field_name] = centered_data_array
-        # import pdb; pdb.set_trace()
         return xr.Dataset(centered_data_dict)
 
     # pylint:disable=too-many-locals
-    def apply_syms(self, new_grid: YeeGrid, sym_center: Coordinate, symmetry: Symmetry):
-        """Create a new AbstractFieldData subclass by interpolating on the supplied ``new_grid``,
-        using symmetries as defined by ``sym_center`` and ``symmetry``."""
+    def apply_syms(self, grid_dict: Dict[str, Coords], sym_center: Coordinate, symmetry: Symmetry):
+        """Create a new SpatialCollectionData subclass by interpolating on the supplied
+        ``new_grid``, using symmetries as defined by ``sym_center`` and ``symmetry``.
+
+        Parameters
+        ----------
+        grid_dict : Dict[str, Coords]
+            Mapping of the data labels in the SpatialCollectionData to new coordinates on which
+            to be interpolated, using symmetries to expand beyond the stored domain.
+        sym_center : Coordinate
+            Position of the symmetry planes in x, y, and z.
+        symmetry : Symmetry
+            Eigenvalues of the symmetry operation in x, y, and z.
+
+        Returns
+        -------
+        SpatialCollectionData
+            A new SpatialCollectionData with the expanded fields. For data labels that are not
+            keys in the grid_dict, the data is returned unmodified.
+        """
 
         new_data_dict = {}
-        yee_grid_dict = new_grid.yee.grid_dict
-        # Defines how field components are affected by a positive symmetry along each of the axes
-        component_sym_dict = {
-            "Ex": [-1, 1, 1],
-            "Ey": [1, -1, 1],
-            "Ez": [1, 1, -1],
-            "Hx": [1, -1, -1],
-            "Hy": [-1, 1, -1],
-            "Hz": [-1, -1, 1],
-        }
 
-        for field, scalar_data in self.data_dict.items():
+        for data_key, scalar_data in self.data_dict.items():
             new_data = scalar_data.data
 
-            # Get new grid locations
-            yee_coords = yee_grid_dict[field].to_list
-
             # Apply symmetries
-            zipped = zip("xyz", yee_coords, sym_center, symmetry)
-            for dim, (dim_name, coords, center, sym) in enumerate(zipped):
-                # There shouldn't be anything to do if there's no symmetry on this axis
-                if sym == 0:
+            zipped = zip("xyz", sym_center, symmetry)
+            for dim, (dim_name, center, sym) in enumerate(zipped):
+                # Continue if no symmetry or the data key is not in the supplied grid_dict
+                if sym == 0 or not grid_dict.get(data_key):
```

Comment:
better to use `grid_dict.get(data_key) is not None` instead of `not grid_dict.get(data_key)`.